src/repository/polling-fetcher.ts

```diff
@@ -0,0 +1,203 @@
+import { EventEmitter } from 'events';
+import { parseClientFeaturesDelta } from '../feature';
+import { get } from '../request';
+import getUrl from '../url-utils';
+import { UnleashEvents } from '../events';
+import { FetcherInterface, FetchingOptions } from './fetcher';
+
+export class PollingFetcher extends EventEmitter implements FetcherInterface {
```

Comment:
this one has all the polling fetching logic I managed to extract from the repository that was overloaded with responsibilities